community/contributing/docs_writing_guidelines.rst

```diff
@@ -277,25 +277,43 @@ How to write methods and classes
 Dynamic vs Static typing
 ~~~~~~~~~~~~~~~~~~~~~~~~
 
-Despite the addition of optional static typing, GDScript remains primarily 
-dynamic language. Unless you are specifily discussing static typing, you
-should only use dynamic typing in your code examples and snipets. 
-This will help reduce confusion to new users by promoting consistent look
-to the code examples they are exposed to. 
+GDScript aims to be concise and easy to learn. Optional static typing conflicts with this vision
+by making documentation more congested. To improve its accessibility for new users, please use 
+dynamic GDScript code samples where possible. The one exception is topics that explain 
+static typing concepts to users.
 
-**Don't** write using using static typing:
+**Don't** add a type hint with a colon or by casting:
 
 ::
 
     var a := 5
     var b: String = "My String"
+    var my_sprite : Sprite = $Sprite as Sprite
 
-**Do** write code using dynamic typing:
+
+**Do** write varibles with dynamic typing:
 
 ::
 
     var a = 5
     var b = "My String"
+    var my_sprite = $Sprite
+
+
+**Don't** write functions with inferred arguments or return types:
+
+::
+
+    func my_func(arg1 : int, arg2 : String = "") -> void:
+    return
```

Comment:
Some best practices:
1. Make sure the body of these method calls is indented one level (on the below example too).
1. If the return type is void, then you don't even need to explicitly `return`.
1. If there is no actual content to a method, the keyword you are supposed to use is `pass`. It explicitly communicates to the user that the method is a stub and needs to be implemented (or that it is a pure virtual function that is meant to be left blank). Writing `return` muddles the intent of the script.

Reply:
I would also recommend having concrete examples in the docs. While developers or technical writers can make sense of `foobar` examples, I find that real-world examples are more accessible to beginners. Plus you can copy-paste them straight from your own projects.

It's not specific to this page, but there's a lot of `var my_thing = random_stuff` in the docs. That's many missed opportunities to show code examples that people could use.